docs/troubleshooting.md

```diff
@@ -0,0 +1,333 @@
+# Troubleshooting
+
+Your AMD sales representative and support team are available to assist with troubleshooting KubernetesOnload
+in your cluster. Please see [release notes](../ReleaseNotes.md).
+
+## FAQ
+
+### How can I confirm Onload Operator has deployed successfully?
+
+[Onload Operator](#onload-operator) resources will be present with their logs clear of errors.
+
+### Why is Onload Operator failing?
+
+Inspect the [Onload Operator](#onload-operator) resources to determine whether:
+
+* allocated namespace (default of `onload-operator-system`) or [selected nodes](#node-host)
+  have insufficient resource quotas or storage availability or supported architectures, or
+* your current user context is insufficiently privileged for creating the cluster-level administrative resources, or
+* `onload-operator` image is not accessible from [registries](#registries) or is a stale image, or
+* the deployment's `DEVICE_PLUGIN_IMG` environment variable is not set, or
+* the deployment, with its Controller Manager container, was deployed but not Onload CRD object, or vice versa.
+
+### How can I confirm my Onload CR has deployed successfully?
+
+* Onload Operator will be deployed (see above).
+* KMM Operator will be a compatible [version](#kmm-operator-version) and [operating](#kmm-operator-operation).
+* [Onload component](#onload-components) resources, including the Onload CR itself,
+  will be applied without validation errors and present in the cluster with their logs clear of errors.
+* Your workloads will be accelerated when they request Onload. See [sfnettest example](config/samples/sfnettest).
+
+### Why is Onload Device Plugin failing?
+
+For example, the pod has a status of `CrashLoopBackOff`.
+
+Use the [Onload component](#onload-components) commands to determine whether:
+
+* `onload-device-plugin`, `onload-worker`, or `onload-user` images failed to pull from [registries](#registries), or
+* the allocated namespace (by default, current context):
+  * has a security policy which disallows privileged pods/roles, or
+  * has insufficient resource quotas, or
+* your current user context is insufficiently privileged for creating privileged resources, or
+* init container failed to copy Onload files to a [node(s) local storage](#node-host) as either:
+  * sufficient writable free space is not available on node filesystem mounts, or
+  * kernel security (eg. SELinux) is lacking a compatible configuration for the mounts, or
+  * one or more of the host mount paths is incompatible with the underlying host filesystem layout, or
+* Onload Worker failed to start an environment for the Onload Control Plane as:
+  * `/dev/onload` was not found due to node host missing:
+    * `onload` kernel module, or
+    * out-of-tree `sfc` kernel module, or
+  * its container ID was not found due to an unsupported CRI, or
+* Onload Device Plugin's AMD Solarflare hardware discovery failed due to unsupported or missing hardware or kernel, or
```

Comment:
What does "kernel" mean in this context? The device plugin doesn't check kernel versions